endpoints/data_objects/src/main.cpp

```diff
@@ -1657,8 +1657,25 @@ namespace
 					}
 
 					if (const auto iter = _args.find("src-resource"); iter != std::end(_args)) {
+						if (_args.find("src-replica-number") != std::end(_args)) {
+							logging::error(
+								*_sess_ptr,
+								"{}: [src-resource] and [src-replica-number] parameters are incompatible.",
+								fn);
+							return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+						}
```

Comment:
The DataObjRepl API enforces that `RESC_NAME_KW` and `REPL_NUM_KW` not be used together, if that's interesting here. Would need to make the `else if` below just an `if` if we wanted to lean on that. What you have here is probably better, but just thought I would mention it.

I moved to the "squash" commits and see that you considered this case as well. :) I guess we can either use this thread to discuss or just resolve this. Up to you.

Reply:
We've discussed this. See https://github.com/irods/irods_client_http_api/pull/454#discussion_r2279738616.